### PR TITLE
Graphql stanza client API

### PR DESCRIPTION
**Pull request by arcusfelis**
Proposed changes include:
* code for client api
* some code from admin api got moved into a helper
* Added mongoose_graphql behaviour with callback for execute/4 function
* Rename Opts to Args to be consistent.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/109783) / small_tests / c862839e3c157fc1784000b1b84b982d862b0778
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3565/109783/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3565/109783/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-28_16.15.02/lib.mongooseim.logs/run.2022-02-28_16.15.02/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/109852) / small_tests / 9637f5adf78b103451b5e79e9410dc4d7bdb1d2f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3565/109852/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3565/109852/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-28_23.54.50/lib.mongooseim.logs/run.2022-02-28_23.54.51/suite.log.html)

**Review by Premwoik — APPROVED** [review]
Looks good to me generally :) I left minor comments

File: src/graphql/user/mongoose_graphql_stanza_user_mutation.erl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+-ignore_xref([execute/4]).
+
+-include("../mongoose_graphql_types.hrl").
+-include("mongoose_logger.hrl").
+-include("jlib.hrl").
+
+-type result() :: {ok, map()} | {error, term()}.
+
+-spec execute(graphql:endpoint_context(), graphql:ast(), binary(), map()) ->
```
[the remark this answers is not in the document]
Comment (arcusfelis) [reply src/graphql/user/mongoose_graphql_stanza_user_mutation.erl:13]: ok

File: priv/graphql/schemas/user/stanza.gql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+type StanzaUserQuery @protected{
+  "Get n last messages to/from a given contact (optional) with limit and optional date"
+  getLastMessages(with: JID, limit: Int = 50, before: DateTime): StanzasPayload
+}
+
+type StanzaUserMutation @protected{
```
[the remark this answers is not in the document]
Comment (arcusfelis) [reply priv/graphql/schemas/user/stanza.gql:12]: done

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/111304) / small_tests / 8b2a5b1c6da5e0c64c4200959a5861e4c4d938d0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3565/111304/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3565/111304/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-03_13.37.35/lib.mongooseim.logs/run.2022-03-03_13.37.35/suite.log.html)

File: src/graphql/mongoose_graphql.erl [hunk 3]
```diff
@@ -19,8 +19,16 @@
                      vars := map(),
                      authorized := boolean(),
                      ctx := map()}.
+-type context() :: map().
+-type object() :: term().
+-type field() :: binary().
+-type args() :: term().
```
Comment (Premwoik) [inline src/graphql/mongoose_graphql.erl:25]: Is it possible for args to be not a `map()` type?

**Review by Premwoik — COMMENTED** [review]
Great! Nice idea with creating this behavior :) I added one comment.

File: src/graphql/mongoose_graphql.erl (hunk 3 above)
Comment (arcusfelis) [reply src/graphql/mongoose_graphql.erl]: ok, probably no.

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/111399) / small_tests / 5507a7814e99db67775b4f8969a91ca29a49b0eb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3565/111399/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3565/111399/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-03-04_08.19.36/lib.mongooseim.logs/run.2022-03-04_08.19.36/suite.log.html)

**Pull request merged**